src/signal_processing_algorithms/energy_statistics/e_divisive.c

```diff
@@ -113,19 +113,20 @@ bool qhat_values(double * diffs, double * qhat_values, int length){
 
     int row;
     for(row=0;row<length;row++) {
-        y_term += square_sum(diffs, length, row, row+1, row, length);
+        y_term += square_sum(distances, length, row, row+1, row, length);
     }
 
     int tau;
     for(tau=0;tau<length;tau++) {
-        qhat_values[tau] = calculate_q(cross_term, x_term, y_term, tau, length - tau);
+        t_stat_values[tau] = calculate_t(cross_term, 2*x_term, 2*y_term, tau, length - tau);
```

Comment:
Why are we doubling the between terms, but not the cross term?

Reply:
x = [[1, 2], [3, 4], [5, 6]]
    y = [[7, 8], [9, 10]]

    Combined matrix is:
    [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]]
    ----------------------------------------------
    Pairwise distances within Combined matrix is:
    ----------------------------------------------
    [
        [ 0.          2.82842712  5.65685425  8.48528137 11.3137085 ]
        [ 2.82842712  0.          2.82842712  5.65685425  8.48528137]
        [ 5.65685425  2.82842712  0.          2.82842712  5.65685425]
        [ 8.48528137  5.65685425  2.82842712  0.          2.82842712]
        [11.3137085   8.48528137  5.65685425  2.82842712  0.        ]
    ]
```
The distance matrix of x is:
[[0.        , 2.82842712, 5.65685425],
 [2.82842712, 0.        , 2.82842712],
 [5.65685425, 2.82842712, 0.        ]]

The distance matrix of y is:
[[0.        , 2.82842712],
 [2.82842712, 0.        ]]

Both of which are symmetric - so we can calculate the sum of just either upper or lower diagonal and double it.

However the distance matrix between x and y is:
[[ 8.48528137, 11.3137085 ],
  [ 5.65685425,  8.48528137],
  [ 2.82842712,  5.65685425]]
which is either the bottom left or top right part of the combined matrix above.

Since it is not symmetric and doesn't need to be accounted twice (there is no "distance from itself" concept here and its just sum of xi-yj), we only need to add the cross term once.
i.e we can use the same cross term that we used for in the earlier implementation of e-divisive (only the way how we calculate pairwise distances WITHIN x and pairwise distances WITHIN y has changed, the pairwise distance BETWEEN the x and y remains same as before).

We can jump on a call if it's not clear.
```

Reply:
Well, the cross term is symmetric over the diagonal (transpose symmetry), but I think the valid point you're making is that we shouldn't be calculating over the whole matrix since to begin with since that would be 2*n*m entries which we then average over n*m --> incorrect!